setup.py

```diff
@@ -26,7 +26,8 @@ def read_file(filename):
     package_data={},
     python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*",
     setup_requires=["setuptools_scm"],
-    install_requires=["click>=6", "six"],
+    install_requires=["click>=6", "pip", "setuptools", "six", "wheel"],
```

Comment:
Actually `setuptools` was removed once in #645 for a reason, but as for https://packaging.python.org/discussions/install-requires-vs-requirements/#install-requires it sounds reasonable

Reply:
I dont mind removing setuptools if that is your request.  Most packaging systems add a dependency on pip->setuptools (even though it is now optional according to pip devs), so it doesnt need to be here explicitly.

However while it may have been removed as an explicit dependency, pip-tools explicitly looks for `setup.py` files, and invokes `distutils.core.run_setup` on them, and 99% of those files will start with an import of `setuptools`.  Without any setuptools, pip-tools is quite broken.

Reply:
> I dont mind removing setuptools if that is your request.

I would rather keep it. Could we add two comments why do we use "click, pip, six" (eg, pip-tools itself uses those packages in code) and "setuptools, wheel" (eg, needs for processing setup.py and working with wheels)?

Reply:
Updated; happy to keep improving it if necessary.

Reply:
Fixed, sorry about that.